### ISSUE TITLE
Extract Outlier Frames not saving in Labeling GUI

### ISSUE THREAD

**Issue by M3K33L** [DeepLabCut/DeepLabCut#2587]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

I am implementing DeepLabCut on both an HPC and my local computer. I have an existing model that I did not use extract outlier frames on. I added new videos to DeepLabCut to test the model's performance on new material. I wanted to refine the machine labels, so I accessed the newly generated files on my local machine via mapping the network drive. I changed the config.yaml to match the local path structure, and opened my updated DeepLabCut GUI from the command line. Then, I open the folder to any of the videos that I just added, and the CollectedData and the machinelabels both load alongside the images. Then, if I move any of the markers, save and close, and reopen, I can see that the marker was back in the same place as before. I have tried changing my path to the network naming convention, the HPC naming convention, and the local naming convention. The only config configuration that works is with the local naming convention; the others do not load into the GUI within the load project option.

Config file (not sure why it has videos and new_videos interspersed):

```
    # Project definitions (do not edit)
Task: Test7
scorer: DL
date: Apr10
multianimalproject: false
identity:

    # Project path (change when moving around)
project_path: Z:/Cluster/Test7-DL-2024-04-10

# /isilon/datalake/maier_lab/Cluster/Test7-DL-2024-04-10/
# Z:/Cluster/Test7-DL-2024-04-10
    # Annotation data set configuration (and individual video cropping parameters)
video_sets:
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 110 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 110 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 111 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 111 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 112 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 112 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/Subject 118 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/106 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/107 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/107 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/108 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/108 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/134 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/78 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/130 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/132 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/78 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/130 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/81 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/134 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/81 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/132 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/78 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/130 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/134 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/81 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/78 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/81 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/134 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/132 Quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/132 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/130 Sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/86 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/87 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/99 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/94 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/86 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/96 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/98 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/89 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/94 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/96 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/97 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/100 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/84 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/99 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/98 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/97 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/87 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/89 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/100 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/videos/84 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/99 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/84 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/87 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/96 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/100 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/96 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/87 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/98 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/86 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/100 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/84 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/89 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/97 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/94 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/89 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/99 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/97 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/86 sucrose.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/94 quinine.mp4:
    crop: 0, 1280, 0, 720
  Z:/Cluster/Test7-DL-2024-04-10/new_videos/98 sucrose.mp4:
    crop: 0, 1280, 0, 720
bodyparts:
- REye
- REyebrowIn
- REyebrowMid
- ReyebrowOut
- LEye
- LEyebrowIn
- LEyebrowMid
- LeyebrowOut
- MidUpperLip
- MidLowerLip
- RMouth
- RCheek
- LMouth
- LCheek
- Nose
- Forehead
- Body
- Chin
- RJaw
- LJaw

start: 0
stop: 1
numframes2pick: 20

    # Plotting configuration
skeleton:
skeleton_color: black
pcutoff: 0.9
dotsize: 12
alphavalue: 0.7
colormap: rainbow

    # Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 2
default_net_type: resnet_50
default_augmenter: default
snapshotindex: -1
batch_size: 8

    # Cropping Parameters (for analysis and outlier frame detection)
cropping: false
    #if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624

    # Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true
```

### Operating System

Local: Windows 11 Enterprise 22H2
HPC: RHEL9

### DeepLabCut version

2.3.9

### DeepLabCut mode

single animal

### Device type

Local: CPU: 12th Gen Intel(R) Core(TM) i7-12700   2.10 GHz
GPU: N/A

HPC: CPU: AMD EPYC 7H12 / 2.6 GHz processor
GPU: NVIDIA A100

### Steps To Reproduce

1. On the HPC, use a slurm batch command to initiate adding new DLC videos to an existing model.
2. Slurm activates environment and python script, which contains the following:

```
#!/home/gadelave/miniconda/envs/DEEPLABCUT/lib python3.9
import tensorflow as tf
import os
import sys

sys.path.append('/home/gadelave/miniconda/envs/DEEPLABCUT/lib/')
sys.path.append('/home/gadelave/miniconda/envs/DEEPLABCUT/lib')
sys.path.append('/home/gadelave/miniconda/envs/DEEPLABCUT/')
sys.path.append('/home/gadelave/miniconda')

os.environ["DLClight"] = "True"

import deeplabcut as dlc

os.chdir('/home/gadelave/miniconda/envs/DEEPLABCUT/lib/')

config_path = '/isilon/datalake/maier_lab/Cluster/Test7-DL-2024-04-10/config.yaml'

new_videos = os.listdir('/isilon/datalake/maier_lab/Cluster/Test7-DL-2024-04-10/new_videos/')
new_videos2 = ['/isilon/datalake/maier_lab/Cluster/Test7-DL-2024-04-10/new_videos/'+video for video in new_videos]
new_video_path = ['/isilon/datalake/maier_lab/Cluster/Test7-DL-2024-04-10/new_videos/']

dlc.add_new_videos(config_path, new_videos2, copy_videos=True)
dlc.analyze_videos(config_path, new_video_path, shuffle=1, save_as_csv=True, videotype='mp4')
dlc.filterpredictions(config_path, new_video_path, shuffle=1, save_as_csv=True, videotype='mp4')
dlc.create_labeled_video(config_path, new_video_path, filtered=True, videotype='mp4', save_frames=False)
dlc.extract_outlier_frames(config_path, new_video_path, automatic=True)
```

4. Map the HPC drive to the local pc to access HPC files
5. Modify config.yaml to match local path naming (Z:/.../Cluster/Test7...) in project name and for each video
6. Conda activate DEEPLABCUT, python -m deeplabcut, load config.yaml
7. In Extract Outlier Frames tab, open labeling GUI
8. In labeling GUI, Files -> Open Folder -> Video1Folder
9. Move machinelabels-iter2 label
10. Save selected layer (ctrl+s)
11. Close napari
12. Does not save after reopening the same folder

### Relevant log output

```shell
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\RES1WM13604P05\AppData\Local\anaconda3\envs\DEEPLABCUT\lib\site-packages\napari\_qt\qt_main_window.py", line 250, in eventFilter
    return QMainWindow.eventFilter(self, source, event)
TypeError: 'PySide6.QtCore.QObject.eventFilter' called with wrong argument types:
  PySide6.QtCore.QObject.eventFilter(QWidgetItem, QEvent)
Supported signatures:
  PySide6.QtCore.QObject.eventFilter(PySide6.QtCore.QObject, PySide6.QtCore.QEvent)
Traceback (most recent call last):
  File "C:\Users\RES1WM13604P05\AppData\Local\anaconda3\envs\DEEPLABCUT\lib\site-packages\napari\_qt\qt_viewer.py", line 953, in _qt_open
    self.viewer.open(
  File "C:\Users\RES1WM13604P05\AppData\Local\anaconda3\envs\DEEPLABCUT\lib\site-packages\napari\components\viewer_model.py", line 1102, in open
    layers = self._open_or_raise_error(
  File "C:\Users\RES1WM13604P05\AppData\Local\anaconda3\envs\DEEPLABCUT\lib\site-packages\napari\components\viewer_model.py", line 1222, in _open_or_raise_error
    raise MultipleReaderError(
napari.errors.reader_errors.MultipleReaderError: Multiple plugins found capable of reading Z:/Cluster/Test7-DL-2024-04-10/labeled-data/78 Quinine. Select plugin from {'napari': 'napari builtins', 'napari-deeplabcut': 'napari DeepLabCut'} and pass to reading function e.g. `viewer.open(..., plugin=...)`.
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by M3K33L** [comment]
We have also attempted to copy the contents of labeled-data to our local pc and then run deeplabcut.refine_labels(config_path) from the command prompt. We changed the config path to the correct location, E:/Documents/Test7-DL-2024-04-10. This resulted in the same non-saving outcome. However, with the use of the command prompt, we were able to extract this error message:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
TypeError: 'PySide6.QtCore.QObject.eventFilter' called with wrong argument types:
  PySide6.QtCore.QObject.eventFilter(QWidgetItem, QEvent)
Supported signatures:
  PySide6.QtCore.QObject.eventFilter(PySide6.QtCore.QObject, PySide6.QtCore.QEvent)
```

Hopefully this helps in determining what might be going on.

**Comment by n-poulsen** [comment]
Hi @M3K33L! I wasn't able to reproduce your issue - I was able to extract outlier frames and save them.

Note that this process is a bit special: when you save labels that you refined, they're saved to the `ColllectedData` layer, and never to the `machinelabels` layer (as specified in the [`napari-deeplabcut` docs](https://github.com/DeepLabCut/napari-deeplabcut/tree/main?tab=readme-ov-file#save-layers)):

- `machinelabels` layer: contains keypoints predicted by your trained model
- `CollectedData` layer: contains keypoints labeled by a human (either from scratch or by refining a `machinelabels` layer)

I recorded how this process happens in the video below: I have a `CollectedData` file laster edited on the 15th of August, and a just-created `machinelabels` file (at 15h48).

At first, the `CollectedData` layer has frame 0 labeled but not frame 1 (as frame 1 was an outlier frame that was extracted). The `machinelabels` layer has  This is illustrated between 10s and 25s in the video. When I correct the `machinelabels` layer and save, the `CollectedData` file is the one updated. Check that you fix **ALL** predictions made in the `machinelabels` file, as when you save the predictions made on all images will be saved to the CollectedData layer. 

https://github.com/user-attachments/assets/321f943a-f04d-49af-8808-84c06b4cbffb

Let me know if this solves your issue!
